docs/proofs/elabReflection.rst

```diff
@@ -0,0 +1,55 @@
+Extending Idris using Elaborator Reflection
+===========================================
+
+Idris provides a mechanism to modify the language without having to recompile Idris itself. We can think of this in terms of metaprogramming or domain specific languages or just building in new capabilities.
+
+In order to extend the language we need to know something about how Idris is complied. On this page we try to limit this explanation to only what we need to customise the elaboration. For more details of the compiler see `Edwin Brady 2013 paper`_ and for customising the elaboration process see `David Christiansen Practical Reflection and Metaprogramming`_ and  `Elaborator reflection: extending Idris in Idris`_. For more information about holes and guesses see `Dependently Typed Functional Programs and their Proofs McBride 1999`_.
+
+Here is a diagram of the multistage process at the top level when Idris code gets compiled:
+
+.. image:: ../image/idrisTopLevel.png
```

Comment:
I've changed some of the arrows, hopefully that's better?

I have also uploaded the SVG source files with instructions for editing.

Reply:
Regarding the antialiasing effect: I am not seeing that on the editors and viewers that I am using. Most importantly I am not seeing it in the html generated by sphinx, which I suspect most readers will use, so I will try to get the lines straight to avoid distracting the reader with things not being straight but I don't think it is a big problem.